content/docs/user-guide/testing/getting-started/_index.md

```diff
@@ -0,0 +1,232 @@
+```
+
+To keep test functions legible at a glance, we recommend using the [Osherove Naming Convention](https://osherove.com/blog/2005/4/3/naming-standards-for-unit-tests.html) of `UnitOfWork_StateUnderTest_ExpectedBehavior`. This can be especially helpful when reading a report with many individual test case failures. In a short a C++ example:
+
+```cpp
+TEST_F(Matrix4x4Tests, MatrixMultiply_InverseMatrix_ReturnIdentityMatrix)
+{
+    // (function under test would be called here)
+    ASSERT_TRUE(someErrorState);
+    EXPECT_TRUE(someResult);
+    EXPECT_FALSE(secondaryProperty);
+}
+```
+
+For more information on writing tests in C++, see [Using AzTest](/docs/user-guide/testing/aztest/aztest/).
+
+## GoogleBenchmark
+
+For performance benchmarks of small pieces of C++ code O3DE uses [GoogleBenchmark](https://github.com/google/benchmark/blob/main/docs/index.md), which is quite similar to GoogleTest. However how failure is defined is a major difference between a normal test and a benchmark. In most tests, pass/fail status is directly evaluated into a boolean state. Benchmarks instead create a subjective performance metric. These metrics are most valuable when periodically recorded, to help detect trends. The only objective failure during a benchmark occurs when the code fails to run or crashes.
+
+The steps for configuring a GoogleBenchmark library are identical to the [steps above for GoogleTest libraries](#googletest), with a couple exceptions. The include statement is `#include <benchmark/benchmark.h>` and the cmake helper function is:
+
+```
+ly_add_googlebenchmark(
+    NAME MyModule.Benchmarks
+)
+```
+
+## PyTest
+
+Certain tests are easier to write in a scripting language like Python. The scope of these tests are typically at the [integration level](https://softwaretestingfundamentals.com/integration-testing/), verifying mid-to-high level functionality of workflows and overall system sanity. Additionally, any Python-based tools will prefer unit tests written in Python. For these tests O3DE uses [PyTest](https://docs.pytest.org/).
+
+While interpreted languages like Python can have low performance, these tests should not perform heavy computation in Python. Tests should instead focus on coordinating workflows: signalling events and then verifying a response. Leave heavy operations to the code targeted by a test.
+
+There are often two Python interpreter instances used during O3DE tests. This can complicate determining which will execute your code, so pay close attention to how the test is defined:
+
+- #1 Exists outside of the O3DE Editor
+  - Use in generic tests which launch applications and send external signals
+  - Helps and monitor for application crashes
+  - Test file names should start with `test_`
+  - Same Python interpreter that launches when you run `o3de/python/python.sh` or `o3de\python\python.cmd`
+
+- #2 Embedded inside the O3DE Editor
+  - Use in "EditorPythonBindings" tests which target workflows inside the editor
```

Comment:
Do we have or plan to have an EPB doc to link to?

Reply:
The "parallel-pattern" doc is all that currently exists for EPB, which is already linked to elsewhere within this guide. I wasn't sure this use also needed a link.

The other guide is currently a bit thin on introducing the system, especially for its non-testing usage.